src/lm.jl

```diff
@@ -243,3 +243,27 @@ function confint(obj::LinearModel; level::Real=0.95)
     hcat(coef(obj),coef(obj)) + stderror(obj) *
     quantile(TDist(dof_residual(obj)), (1. - level)/2.) * [1. -1.]
 end
+
+"""
+    cooksdistance(obj::LinearModel)
+
+Compute [Cook's distance](https://en.wikipedia.org/wiki/Cook%27s_distance)
+for each observation in linear model `obj`, giving an estimate of the influence
+of each data point.
+Currently only implemented for LinearModel models without weights.
+"""
+function StatsBase.cooksdistance(obj::LinearModel)
+    u = residuals(obj)
+    mse = dispersion(obj,true)
+    k = dof(obj)-1
+    d_res = dof_residual(obj)
+    X = modelmatrix(obj)
+    wts = obj.rr.wts
+    if isempty(wts)
+        hii = diag(X * inv(X' * X) * X')
```

Comment:
This is a pretty expensive operation for computing the diagonal of the hat matrix. I need to write out some algebra to optimize this, but basically anytime you saw an inverse of a matrix of arbitrary size and dimension in numeric code, something has gone wrong. (channeling @dmbates here in my attitudes, but maybe he has the formula handy without needing to write some scribbles involving a Cholesky.....)

Reply:
Actually, I wondering now if we should define (maybe in StatsBase?)

1. `hatmatrix` to match `crossmodelmatix`.
2. `diaghatmatrix` that is an efficient evaluation of *only* the diagonal of the hat matrix.

(thoughts @ararslan and @nalimilan ?)

Reply:
Okay, assuming that `X` is full rank (i.e. there are no collinear predictors, which your code already does with `inv`), then you can use the QR decomposition to compute the hat matrix more efficiently and in a numerically stable way:

```julia
julia> using LinearAlgebra

julia> X = [1 0; 1 1; 1 0; 1 1]
4×2 Matrix{Int64}:
 1  0
 1  1
 1  0
 1  1

julia> hat = X * inv(X'X) * X'
4×4 Matrix{Float64}:
 0.5  0.0  0.5  0.0
 0.0  0.5  0.0  0.5
 0.5  0.0  0.5  0.0
 0.0  0.5  0.0  0.5

julia> Q = qr(X).Q[1:size(X,1), 1:size(X,2)]
4×2 Matrix{Float64}:
 -0.5   0.5
 -0.5  -0.5
 -0.5   0.5
 -0.5  -0.5

julia> Q*Q'
4×4 Matrix{Float64}:
 0.5          5.55112e-17  0.5          8.32667e-17
 5.55112e-17  0.5          8.32667e-17  0.5
 0.5          8.32667e-17  0.5          1.11022e-16
 8.32667e-17  0.5          1.11022e-16  0.5

julia> round.(Q*Q'; digits=1)
4×4 Matrix{Float64}:
 0.5  0.0  0.5  0.0
 0.0  0.5  0.0  0.5
 0.5  0.0  0.5  0.0
 0.0  0.5  0.0  0.5

julia> hat - Q*Q' .< eps(Float64) # everything within machine epsilon 
4×4 BitMatrix:
 1  1  1  1
 1  1  1  1
 1  1  1  1
 1  1  1  1
```

Reply:
Well, my intention was to write some tests to finalize the implementation of Cook's Distance. 
It feels to me that the optimizations you're suggestions would optimize the code, but I am not sure where to start as to code it or to test it, nor in which package it makes the most sense to introduce it.
Could this be done in subsequent PR, by someone more skilled than me in optimizing the code and with a better knowledge of the intricacies of GLM, StatsModel, and StatsBase?

Reply:
@palday well the collinearity is still an issue. It is dealt with only in the sense that invoking the calculation under the wrong condition provides an error.

Reply:
Thanks @palday. I suspected there were better ways to do this. Though by default we use the Cholesky decomposition, so I guess we'd better whatever decomposition has already been computed (and stored) in the predictor object?

Reply:
@nalimilan Yep, I found some notes for dealing with QR or Cholesky efficiently.